plugin/StrikePluginHostedService.cs

```diff
@@ -0,0 +1,194 @@
+	{
+		_strikeApiListenLoop = StartStrikeApiLoop();
+		Subscribe<InvoiceEvent>();
+		base.SubscribeToEvents();
+	}
+
+	private async Task? StartStrikeApiLoop()
+	{
+		var cancellation = _cts.Token;
+		while (!cancellation.IsCancellationRequested)
+		{
+			try
+			{
+				await cleanupOldExpiredQuotes();
+				
+				await using var db = _dbContextFactory.CreateContext();
+				
+				// update quotes that are waiting for payment
+				var waitingQuotes = await db.Quotes.Where(a => !a.Observed && !a.Paid).ToArrayAsync(cancellation);
+
+				var distinctTenants = waitingQuotes.Select(b => b.TenantId).Distinct().ToArray();
+				foreach (var tenantId in distinctTenants)
+				{
+					var tenantQuotes = waitingQuotes.Where(a => a.TenantId == tenantId).ToArray();
+					var client = _hodler.GetClient(tenantId);
+					
+					// clients may still not be initialized during BTCPayServer startup, so we can wait for them to be init 
+					if (client == null)
+						continue;
+
+					await updateInvoicesForTenant(client, tenantQuotes);
+				}
+
+				await Task.Delay(1000, cancellation);
+			}
+			catch when (cancellation.IsCancellationRequested)
+			{
+			}
+			catch (Exception ex)
+			{
+				_logger.LogError(ex, "StrikePluinHostedService error: "+ ex);
+			}
+		}
+	}
+
+	// on old instances, we may have quotes that are expired but not observed... leaving this for extreme cases
+	private async Task cleanupOldExpiredQuotes()
+	{
+		await using var db = _dbContextFactory.CreateContext();
+		
+		var expiredQuotes = db.Quotes.Where(a => 
+			!a.Observed && !a.Paid && a.ExpiresAt < DateTimeOffset.UtcNow);
+		foreach (var expired in expiredQuotes)
+			expired.Observed = true;
+                
+		await db.SaveChangesAsync(_cts.Token);
+	}
+
+	private async Task updateInvoicesForTenant(StrikeLightningClient client, StrikeQuote[] quotes)
+	{
+		var invoices = await queryStrikeApi(client, quotes);
+		
+		await using var db = _dbContextFactory.CreateContext();
+		foreach (var invoice in invoices.Where(a => a.State == InvoiceState.Paid))
+		{
+			var quote = await db.Quotes.SingleOrDefaultAsync(a => a.InvoiceId == invoice.InvoiceId.ToString());
+			if (quote == null)
+				continue;
+
+			quote.Paid = true;
+
+			// saving changes so that invoice listener can immediately pick up on this
+			await db.SaveChangesAsync();
+
+			// PROCESS CURRENCY CONVERSION
+			if (quote.PaidConvertTo != null)
+			{
+				var convertTo = (Currency)Enum.Parse(typeof(Currency), quote.PaidConvertTo, true);
+				var target = (Currency)Enum.Parse(typeof(Currency), quote.TargetCurrency, true);
+				var req = new CurrencyExchangeQuoteReq
+				{
+					Sell = target,
+					Buy = convertTo,
+					Amount = new CurrencyExchangeAmount
+					{
+						Currency = target, Amount = quote.TargetAmount, FeePolicy = FeePolicy.Exclusive
+					}
+				};
+
+				var success = await client.ExecCurrencyConversion(req, _cts.Token);
+				if (success)
+				{
+					quote.PaidConvertTo = null;
```

Comment:
We can use `idempotent key` in `CurrencyExchangeQuoteReq` to avoid double execution and keep this `PaidConvertTo` value in the DB (can be useful for debug or problem analysis).